### DISCUSSION TITLE
80,000 queued messages which I want to cancel - how?

### DISCUSSION THREAD

**Discussion by chrisdeeming** [postalserver/postal#2884] [Help with using Postal] [answered]
A user of our mail server currently has over 80,000 messages queued which has flooded the server to the point that legitimate mails are not being sent.

What is the recommended approach to deal with this. Happy to just delete all messages from a specific address.

Would this work?

```sql
DELETE FROM messages WHERE mail_from = 'example@example.com' AND status = 'Pending';
```

Or do I need to clear them from the queued_messages table? 

Or is there a better approach?

Sorry to ping you @willpower232 and @adamcooke but I feel like you guys would know (fairly urgent).

**Comment by adamcooke** [comment, marked as answer]
You would need to remove messages from the `queued_messages` table if you want the queue to be processed faster. The actual `messages` table isn't so important.

**Comment by chrisdeeming** [comment]
Thanks @adamcooke. I appreciate the quick reply! Thank you!

In the end I did:

```sql
DELETE FROM `postal`.queued_messages WHERE id IN (SELECT id FROM `postal-server-1`.messages WHERE mail_from = 'email' AND status = 'Pending' ORDER BY timestamp DESC);
```

Followed by:

```sql
DELETE FROM `postal-server-1`.messages WHERE mail_from = 'email' AND status = 'Pending' ORDER BY timestamp DESC;
```

Please let me know if you envisage anything else I should do to eradicate anything leftover by this and avoid causing any issues.

**Reply by adamcooke** [reply]
There will be data in the raw messages tables too but they'll clear up on own after 30 days.

**Comment by chrisdeeming** [comment]
As a side note, which config parameters control how many queued messages are sent at a time? We're left with over 10,000 messages that were delayed by the previous queue. The server itself should be able to cope with pushing these out quicker than it seems to be right now.

**Reply by adamcooke** [reply]
What version are you running? With v3 you can add more workers to increase capacity to process the queue faster. Additionally, more CPU cores and better optimised MySQL will also be able to help improve throughput speed.

**Reply by chrisdeeming** [reply]
Yeah latest v3 release

**Reply by chrisdeeming** [reply]
Looks like this might help?

https://github.com/orgs/postalserver/discussions/2879#discussioncomment-8800758

**Reply by chrisdeeming** [reply]
```
# postal start
[+] Running 6/6
 ✔ Container postal-worker-4  Started                                                                                                               0.1s 
 ✔ Container postal-web-1     Started                                                                                                               0.1s 
 ✔ Container postal-worker-1  Started                                                                                                               0.1s 
 ✔ Container postal-worker-2  Started                                                                                                               0.1s 
 ✔ Container postal-worker-3  Started                                                                                                               0.1s 
 ✔ Container postal-smtp-1    Started                                                                                                               0.1s
```

💪
